api.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/*Doccumentation for how use this REST API is included in the sumbitnums.html file*/
+    header("Access-Control-Allow-Methods: OPTIONS,GET,POST,PUT,DELETE");
+    header("Content-Type:application/json"); //sends the http json header to browser
+    function connect(){
+        //Connection to my database
+        $dbServername = "localhost";
+        $dbUsername = "nurudeen_agb";
+        $dbPassword = "head.storm";
+        $dbName = "nurudeen_headstormapi";
+        $con = mysqli_connect($dbServername,$dbUsername,$dbPassword,$dbName);
+        //checking if connection to the database was successful
+        if(!$con){
+            die("Connection failed: ". mysqli_connect_error());
+        }
+        else
+        print_r ("Connection successful");
+        echo nl2br("\r\n");
+        return $con;
+    }
+    
+    /*This function retrieves the requsted data from the database and returns the sorted list of numbers*/
+    function readsortedfromDB($listName){
+        $con = connect();
+        $sql = "SELECT * FROM posts WHERE listName = '$listName';";
+        $result = mysqli_query($con,$sql);
+        if( $result){
+           echo "      Succesful";
+       }
+       else{
+           echo "Error " .$sql . "<br>" .mysqli_error($con);
+       }
+        $jsonResult = mysqli_fetch_assoc($result);
+        //print_r($jsonResult);
+        $post = json_decode($jsonResult['nums'], TRUE);
+        $nums = $post['nums'];
+        sort($nums);
```

Comment:
Some things I could do;

- Delete the lines where I had 'echo "GET works"' and 'echo "POST works"'. I used these lines of code for debugging and I forgot to discard these before submitting

- I can use single quotes instead of double quotes when using echo on strings that don't have any variables like the line of code > echo " List does not exist"

- For the GET case in the switch statement, I can pass the list name variable by reference >readsortedfromDB(&$listName)  rather than by value >readsortedfromDB($listName) for better performance

- I can use caching techniques to store a few previously requested lists to reduce the load of database operations

- To save memory I can unset $_POST["listName] superglobal after it is called

- Make $con a global variable so I don't have to create a new database connection for every time I call readsortedfromDB